src/dac.rs

```diff
@@ -0,0 +1,171 @@
+//! Configure the internal DAC on the stm32f3xx.
+//! Incomplete, but includes basic operation.
+
+use crate::{
+    gpio::{
+        gpioa::{PA4, PA5},
+        Analog,
+    },
+    pac,
+    rcc::APB1,
+};
+use core::convert::Infallible;
+
+/// Trait representing a single-channel digital-to-analog converter (DAC).
+pub trait SingleChannelDac<Word> {
+    /// Error type returned by DAC methods
+    type Error;
+
+    /// Output a constant signal, given a bit word.
+    fn try_set_value(&mut self, value: Word) -> Result<(), Self::Error>;
+}
+
+/// This is an abstraction to select the correct DAC channel for a given
+/// (analog) output pin.
+pub trait Pin {
+    const CHANNEL: Channel;
+}
+
+impl Pin for PA4<Analog> {
+    const CHANNEL: Channel = Channel::One;
+}
+
+impl Pin for PA5<Analog> {
+    const CHANNEL: Channel = Channel::Two;
+}
+
+#[derive(Clone, Copy)]
+/// Select the channel
```

Comment:
Really minor nit. In all other modules documentation string come before derive annotations.